src/CMakeLists.txt

```diff
@@ -1,295 +1,296 @@
+    ${CMAKE_SOURCE_DIR}/src/name.h
+    ${CMAKE_SOURCE_DIR}/src/tile_id_data.h
+    ${CMAKE_SOURCE_DIR}/src/crafting.h
+    ${CMAKE_SOURCE_DIR}/src/output.h
+    ${CMAKE_SOURCE_DIR}/src/material.h
+    ${CMAKE_SOURCE_DIR}/src/action.h
+    ${CMAKE_SOURCE_DIR}/src/dependency_tree.h
+    ${CMAKE_SOURCE_DIR}/src/martialarts.h
+    ${CMAKE_SOURCE_DIR}/src/iexamine.h
+    ${CMAKE_SOURCE_DIR}/src/iuse_software_sokoban.h
+    ${CMAKE_SOURCE_DIR}/src/options.h
+    ${CMAKE_SOURCE_DIR}/src/get_version.h
+    ${CMAKE_SOURCE_DIR}/src/mapsharing.h
+    ${CMAKE_SOURCE_DIR}/src/catalua.h
+    ${CMAKE_SOURCE_DIR}/src/trap.h
+    ${CMAKE_SOURCE_DIR}/src/item_factory.h
+    ${CMAKE_SOURCE_DIR}/src/weather_gen.h
+    ${CMAKE_SOURCE_DIR}/src/platform_win.h
+    ${CMAKE_SOURCE_DIR}/src/active_item_cache.h
+    ${CMAKE_SOURCE_DIR}/src/construction.h
+    ${CMAKE_SOURCE_DIR}/src/player.h
+    ${CMAKE_SOURCE_DIR}/src/advanced_inv.h
+    ${CMAKE_SOURCE_DIR}/src/pldata.h
+    ${CMAKE_SOURCE_DIR}/src/start_location.h
+    ${CMAKE_SOURCE_DIR}/src/inventory.h
+    ${CMAKE_SOURCE_DIR}/src/gamemode.h
+    ${CMAKE_SOURCE_DIR}/src/artifact.h
+    ${CMAKE_SOURCE_DIR}/src/mapdata.h
+    ${CMAKE_SOURCE_DIR}/src/text_snippets.h
+    ${CMAKE_SOURCE_DIR}/src/monattack.h
+    ${CMAKE_SOURCE_DIR}/src/filesystem.h
+    ${CMAKE_SOURCE_DIR}/src/player_activity.h
+    ${CMAKE_SOURCE_DIR}/src/addiction.h
+    ${CMAKE_SOURCE_DIR}/src/iuse_actor.h
+    ${CMAKE_SOURCE_DIR}/src/wcwidth.h
+    ${CMAKE_SOURCE_DIR}/src/uistate.h
+    ${CMAKE_SOURCE_DIR}/src/profession.h
+    ${CMAKE_SOURCE_DIR}/src/path_info.h
+    ${CMAKE_SOURCE_DIR}/src/version.h
+    ${CMAKE_SOURCE_DIR}/src/input.h
+    ${CMAKE_SOURCE_DIR}/src/disease.h
+    ${CMAKE_SOURCE_DIR}/src/item_stack.h
+    ${CMAKE_SOURCE_DIR}/src/moraledata.h
+    ${CMAKE_SOURCE_DIR}/src/itype.h
 )
 
 # Add the actual executable
-add_executable(cataclysm ${SRCS})
+ADD_EXECUTABLE(cataclysm ${CATACLYSM_DDA_SOURCES} ${CATACLYSM_DDA_HEADERS})
+
+FIND_PACKAGE(Git)
 
-# Generate version information
-if (WIN32)
-	add_custom_target(gen_version
-		COMMAND for /F \"tokens=*\" %%i in \('git describe --tags --always --dirty --match \"[0-9]*.[0-9]*\"'\) do set VERSION=%%i
-		COMMAND echo \#define VERSION \"%VERSION%\" > ${CMAKE_SOURCE_DIR}/src/version.h
-	)
-else()
-	add_custom_target(gen_version
-		COMMAND echo -n \"\#define VERSION \" > ${CMAKE_SOURCE_DIR}/src/version.h
-		COMMAND git describe --tags --always --dirty --match \"[0-9]*.[0-9]*\" >> ${CMAKE_SOURCE_DIR}/src/version.h
-	)
-endif()
-add_dependencies(cataclysm gen_version)
+ADD_CUSTOM_TARGET(
+	get_version
+	${CMAKE_COMMAND} -D SRC=${CMAKE_SOURCE_DIR}/src/version.h.in
+	-D DST=${CMAKE_SOURCE_DIR}/src/version.h
+	-P ${CMAKE_SOURCE_DIR}/src/version.cmake
+)
+
+ADD_DEPENDENCIES(cataclysm get_version)
+IF (LUA)
+	ADD_DEPENDENCIES(cataclysm lua_bindings)
+ENDIF(LUA)
 
 # Required library linkage
-if(TILES)
+IF(TILES)
 	target_include_directories(cataclysm PUBLIC ${SDL2_INCLUDE_DIR})
```

Comment:
Since the build was broken before and nobody noticed in 15 days its probably safe to upgrade.